youtube_dl/extractor/kontrtube.py

```diff
@@ -31,16 +31,25 @@ class KontrTubeIE(InfoExtractor):
 
     def _real_extract(self, url):
         mobj = re.match(self._VALID_URL, url)
-        video_id = mobj.group('id')
-        display_id = mobj.group('display_id')
+        video_id, display_id = mobj.groups()
 
-        webpage = self._download_webpage(
-            url, display_id, 'Downloading page')
+        webpage = self._download_webpage(url, display_id)
+
+        video_data = self._parse_json(
+            self._search_regex(
+                r'var\s+flashvars\s*=\s*({[^}]+});', webpage, 'video data', fatal=False),
+            video_id, transform_source=js_to_json)
```

Comment:
You seem misunderstand the point completely. This piece of code will break extraction if failed and it can fail easily due to inner dict in JSON (since you don't allow `}`), due to parts `js_to_json` can't fix or due to some other reason that will eventually produce unparsable JSON.